gtk/OptionsDialog.cc

```diff
@@ -351,6 +350,18 @@ void TorrentFileChooserDialog::onOpenDialogResponse(int response, Glib::RefPtr<S
 #else
         auto const files = get_files();
 #endif
+        /* remember this folder the next time we use this dialog */
+        if (auto const folder = IF_GTKMM4(get_current_folder, get_current_folder_file)(); folder != nullptr)
+        {
+            gtr_pref_string_set(TR_KEY_open_dialog_dir, folder->get_path());
+        }
+        else if (!files.empty())
```

Comment:
Honestly, I thought the same thing, but I'm not aware of preferred way to assert in GTK client.
Can you suggest right way to do it?
IMO, this assert with some informative debug message should be present in release code, as calling `front()` on an empty vector is UB.

Reply:
If you want it to work in release mode too (and print a somewhat nice message), I'd use

```cpp
g_assert_false(files.empty());
```

Otherwise (for debug only),

```cpp
g_assert(!files.empty());
```

Either way it fine by me.